eyelid-match-ops/src/primitives/poly.rs

```diff
@@ -326,3 +330,76 @@ pub fn poly_split_half(a: &Poly, chunk: usize) -> (Poly, Poly) {
 
     (remainder, quotient)
 }
+
+/// Returns the inverse in the cyclotomic ring, if it exists.
+/// Otherwise, returns an error.
+pub fn inverse(a: &Poly) -> Result<Poly, String> {
+    let mut mod_pol = Poly::zero();
+
+    // TODO: don't recompute the modulus here
+    mod_pol[MAX_POLY_DEGREE] = Coeff::one();
+    mod_pol[0] = Coeff::one();
+
+    extended_gcd(&mod_pol, a)
+}
+
+/// Helps to calculate the equation `cur = prev - q.cur`.
+fn update_diophantine(prev: Poly, cur: Poly, q: Poly) -> (Poly, Poly) {
+    let aux = cur.clone();
+    let mul_res = cur.mul(&q.clone());
+    let new_cur = prev.sub(&mul_res).clone();
+    let new_prev = aux.clone();
+
+    (new_cur, new_prev)
+}
```

Comment:
After implementing some missing operators, this can be simplified, and tweaked to re-use most of the supplied polynomials.